tests/integration/test_integration.py

```diff
@@ -770,3 +898,72 @@ async def run_restore_backup_action(
 
 def copy_lib_content() -> None:
     shutil.copyfile(src=VAULT_KV_LIB_DIR, dst=f"{VAULT_KV_REQUIRER_CHARM_DIR}/{VAULT_KV_LIB_DIR}")
+
+
+async def read_vault_unit_statuses(ops_test: OpsTest):
+    status_tuple = await ops_test.juju("status")
+    if status_tuple[0] != 0:
+        raise Exception
+    output = []
+    for row in status_tuple[1].split("\n"):
+        if not row.startswith(f"{APPLICATION_NAME}/"):
+            continue
+        cells = row.split(maxsplit=4)
+        if len(cells) < 5:
+            cells.append("")
+        output.append(
+            {
+                "unit": cells[0],
+                "status": cells[1],
+                "agent": cells[2],
+                "address": cells[3],
+                "message": cells[4],
+            }
+        )
+    return output
+
+
+async def wait_for_vault_status_message(
+    ops_test: OpsTest, count: int, expected_message: str, timeout: int = 100, cadence: int = 5
+):
+    while timeout > 0:
```

Comment:
There are two methods that do similar timeouts here, let's standardize with 1 approach (see `_get_vault_traefik_endpoint` which does it differently)

Reply:
Didn't even notice that. Moved them closer together and updated both, also added docstring.